src/opendr/perception/object_detection_2d/gem/gem_learner.py

```diff
@@ -0,0 +1,1059 @@
+            ),
+            iters=10,
+            dataset=None,
+            lr=1e-4,
+            batch_size=1,
+            optimizer="adamw",
+            backbone="resnet50",
+            checkpoint_after_iter=0,
+            checkpoint_load_iter=0,
+            temp_path="temp",
+            device="cuda",
+            threshold=0.7,
+            num_classes=91,
+            return_segmentations=False,
+    ):
+
+        # Pass the shared parameters on super's constructor so they can get initialized as class attributes
+        super(GemLearner, self).__init__(
+            iters=iters,
+            lr=lr,
+            batch_size=batch_size,
+            optimizer=optimizer,
+            backbone=backbone,
+            checkpoint_after_iter=checkpoint_after_iter,
+            checkpoint_load_iter=checkpoint_load_iter,
+            temp_path=temp_path,
+            device=device,
+            threshold=threshold,
+        )
+
+        # Add arguments to a structure like in the original implementation
+        self.args = utils.load_config(model_config_path)
+        self.datasetargs = utils.load_config(dataset_config_path)
+        self.args.backbone = self.backbone
+        self.args.device = self.device
+        self.args.num_classes = num_classes
+        self.args.dataset_file = "coco"
+
+        self.dataset = dataset
+        if self.dataset is not None:
+            print("True")
+            self.deleteme()
+
+        if return_segmentations:
+            self.args.masks = True
+            self.args.dataset_file = "coco_panoptic"
+        else:
+            self.args.masks = False
+            self.args.dataset_file = "coco"
+
+        # Initialise distributed mode in case of distributed mode
+        utils.init_distributed_mode(self.args)
+        if self.args.frozen_weights is not None:
+            assert self.args.masks, "Frozen training is meant for segmentation only"
+
+        # Fix the seed for reproducibility
+        seed = self.args.seed + utils.get_rank()
+        torch.manual_seed(seed)
+        np.random.seed(seed)
+        random.seed(seed)
+
+        # Initialise epoch
+        self.epoch = self.checkpoint_load_iter
+
+        # Initialise transform for inference
+        self.infer_transform = T.Compose([
+            T.Resize(self.args.input_size),
+            T.ToTensor(),
+            T.Normalize(self.args.image_mean, self.args.image_std)
+        ])
+
+        # Initialise temp path
+        if not os.path.exists(self.temp_path):
+            os.makedirs(self.temp_path)
+
+        # Initialise ort
+        self.ort_session = None
+
+        # Initialize criterion, postprocessors, optimizer and scheduler
+        self.criterion = None
+        self.postprocessors = None
+        self.torch_optimizer = None
+        self.lr_scheduler = None
+        self.n_parameters = None
+        self.fusion_method = 'sc_avg'
+
+    def save(self, path, verbose=False):
+        """
```

Comment:
This is a minor comment, since in the specification we haven't explicitly stated that docstrings will be used for documentation (given that we separately document the code). However, I think that we should try to be as consistent as we can. Please have a look on the other implementations regarding the comment style:

https://github.com/tasostefas/opendr_internal/blob/433b93f90b90009a63a9d23cc5944e20bc3b6245/src/opendr/perception/heart_anomaly_detection/gated_recurrent_unit/gated_recurrent_unit_learner.py#L157

https://github.com/tasostefas/opendr_internal/blob/433b93f90b90009a63a9d23cc5944e20bc3b6245/src/opendr/perception/object_detection_2d/ssd/ssd_learner.py#L116

https://github.com/tasostefas/opendr_internal/blob/433b93f90b90009a63a9d23cc5944e20bc3b6245/src/opendr/perception/object_tracking_2d/fair_mot/object_tracking_2d_fair_mot_learner.py#L137

Reply:
I know see that I was using a different format for the docstrings, I will adjust it to make it more consistent.